sklearn/metrics/tests/test_pairwise.py

```diff
@@ -331,6 +333,47 @@ def test_pairwise_parallel(func, metric, kwds, dtype):
     assert_allclose(S, S2)
 
 
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "func, metric, kwds",
+    [
+        (pairwise_distances, "euclidean", {}),
+        (
+            pairwise_distances,
+            minkowski,
+            _minkowski_kwds,
+        ),
+        (
+            pairwise_distances,
+            "minkowski",
+            _minkowski_kwds,
+        ),
+        (pairwise_kernels, "polynomial", {"degree": 1}),
+        (pairwise_kernels, callable_rbf_kernel, {"gamma": 0.1}),
+    ],
+)
+def test_pairwise_parallel_array_api(
+    func, metric, kwds, array_namespace, device, dtype_name
+):
+    with config_context(array_api_dispatch=True):
+        xp = _array_api_for_tests(array_namespace, device)
+        rng = np.random.RandomState(0)
+        X_np = np.array(5 * rng.random_sample((5, 4)), dtype=dtype_name)
+        Y_np = np.array(5 * rng.random_sample((3, 4)), dtype=dtype_name)
+        X = xp.asarray(X_np, device=device)
+        Y = xp.asarray(Y_np, device=device)
```

Comment:
Regarding creating `X` and `Y` I am getting a paradox... 

If I were to do `X = xp.asarray(X_np, dtype=dtype_name, device=device)` the tests complain about `dtype_name` being a string when a `dtype` is expected. 

But if I were to do the trick with `X_np` then cast it into the `xp` namespace, the `dtype_name` being passed into must be in the `np` namespace and that defeats the purpose of this test 

I feel like there must be a way around this, and I would appreciate any advice that may point me to the right direction!

Reply:
> But if I were to do the trick with X_np then cast it into the xp namespace, the dtype_name being passed into must be in the np namespace and that defeats the purpose of this test.

I don't understand why you think that defeats the purpose of this test.

Alternatively you can do:

```python
X_np = 5 * rng.random_sample((5, 4))
dtype = getattr(xp, dtype_name)
X_xp = xp.asarray(X_np, device=device, dtype=dtype)
```

but that should yield the same result in the end.